# Taller 1

## Integrantes

* Laura Sofia Fernandez | A00410462
* Juan Miguel Correa | A00412937
* Juan Jose Navas | A00411769
* Juan Camilo Libreros | A00412339


### Pregunta 1

> Despeje de \( r \) en la fórmula del área lateral del cilindro:

$R =  \frac{A}{2\pi g}$

>Despeje de \( h \) en la fórmula del volumen del cono:


$ h = \frac{3V}{A_B} $

>Despeje de \( r \) en la fórmula del volumen de la esfera:

$ r = \sqrt[3]{\frac{3V}{4\pi}} $

>Despeje de \( h \) en la fórmula del área del trapecio:

$ h =  \frac{2A}{B + b} $

>Despeje de \( t \) en la fórmula de la intensidad eléctrica:

$ t =  \frac{q}{I} $

>Despeje de \( R \) en la fórmula de la ley de Ohm:

$ R =  \frac{V}{I} $

>Despeje de \( t \) en la ecuación de la energía eléctrica:

$ E =  \frac{E 3 R}{V^2} $

>Despeje de \( R \) en la ecuación de resistencias en paralelo:

$ R = \frac{R_1 \cdot R_2}{R_1 + R_2} $


> Despeje de \( P_f \) en la ecuación:

$ P_f = \frac{P_0 V_0 T_f}{T_0 V_f} $


> Despeje de \( V_B \) en la ecuación:

$ V_B =  V_A - I R $

### Pregunta 2

A continuación, encontraran una serie de operaciones aritméticas que
deben ser resueltas y comprobadas con Python. Tenga cuidado con la
jerarquía de *operaciones*


• $ [3(−5 + 10 + (−5)) − 2((−2) + (−4) + 10)]^2 $

• $ 45 + [3 (2 + 8) – \frac{8}{4}] + 10 − 83 $

• $ [( −4 + 7)2 − 10]. (−4)– 7 – 3 $

In [ ]:
# Primera expresión
exp1 = (3 * (-5 + 10 + (-5)) - 2 * ((-2) + (-4) + 10)) ** 2

# Segunda expresión
exp2 = 45 + (3 * (2 + 8) - 8 / 4) + 10 - 83

# Tercera expresión
exp3 = ((-4 + 7) * 2 - 10) * (-4) - 7 - 3

print("Resultado primera expresión","=",exp1)
print("Resultado primera expresión","=",exp2)
print("Resultado primera expresión","=",exp3)

### Pregunta 3

Las siguientes operaciones deben ser escritas y resultas en Python



In [9]:
operacion_1= (1/5 * ((1/2)**2 + (3/4)**2) + 10/25 * ((1/2)**2 + (3/4)**2) + 8/40 * ((1/2)**2 + (3/4)**2))
operacion_2= (9/4 * (6/8)**-1 - 5/2 * (25/20)**-1 + 40/8 * (20/16)**-1)

print(operacion_1)
print(operacion_2)

0.65
5.0


### Pregunta 4
 Crea un programa que pida al usuario Nombre, Peso y Talla y este calcule el
IMC



In [ ]:
# funcion que calcula el imc
def calcular_imc(nombre, peso, altura):
  imc = peso / (altura ** 2)
  print(f"Hola {nombre}, tu IMC es: {imc:.2f}")

# funcion principal
def main():
    nombre = input("Ingrese su nombre: ")
    peso = float(input("Ingrese su peso en kilogramos: "))
    altura = float(input("Ingrese su altura en metros: "))
    calcular_imc(nombre, peso, altura)


if __name__ == "__main__":
  main()


Ingrese su nombre: Miguel
Ingrese su peso en kilogramos: 100
Ingrese su altura en metros: 1.90
Hola Miguel, tu IMC es: 27.70


### Pregunta 5
Escriba un programa que pida dos números luego se imprima la suma,
resta, producto y cociente



In [10]:
# Funcion para hacer la suma, resta, producto y division
def calc(num_1, num_2):
  suma = num_1 + num_2
  resta = num_1 - num_2
  producto = num_1 * num_2
  cociente = num_1 / num_2

  print(f"La suma de {num_1} y {num_2} es: {suma}")
  print(f"La resta de {num_1} y {num_2} es: {resta}")
  print(f"El producto de {num_1} y {num_2} es: {producto}")
  print(f"El cociente de {num_1} y {num_2} es: {cociente:.2f}")


# funcion principal
def main():
    num_1 = float(input("Ingrese el primer número: "))
    num_2 = float(input("Ingrese el segundo número: "))
    calc(num_1, num_2)


if __name__ == "__main__":
  main()


Ingrese el primer número: 5
Ingrese el segundo número: 10
La suma de 5.0 y 10.0 es: 15.0
La resta de 5.0 y 10.0 es: -5.0
El producto de 5.0 y 10.0 es: 50.0
El cociente de 5.0 y 10.0 es: 0.50


### Pregunta 6

Construya una lista que Almacene el precio del Barril de Petróleo los
últimos 7 días. Indague sobre el precio de un litro de Cocacola (Hoy)
y haga la conversión de Litros a Barriles bbl oil. Construya otra lista
que relacione proporcionablemente el precio de un barril de Petróleo
vs un barril de Cocacola. Construya un párrafo con sus conclusiones

In [3]:
# Lista del precio del barril de petróleo (en USD) durante los últimos 7 días
precio_petroleo = [76.24, 77.88, 78.31, 77.22, 76.71, 76.45, 77.17]

# Precio de un litro de Coca‑Cola (dato aproximado en USD)
precio_litro_coc = 1.50

# Conversión: 1 barril de petróleo equivale a 159 litros
conversion_barril = 159

# Precio de un "barril" de Coca‑Cola (convertido a partir del precio de 1 litro)
barril_coc = precio_litro_coc * conversion_barril

# Lista que relaciona el precio del petróleo con el precio de un "barril" de Coca‑Cola
# (se calcula el cociente: precio del petróleo / precio del "barril" de Coca‑Cola)

ratio_list = [round(price / barril_coc, 3) for price in precio_petroleo]


print("Precios del barril de petróleo (últimos 7 días):", precio_petroleo)
print("Precio de un 'barril' de Coca‑Cola (159 litros):", barril_coc)
print("Relación (Petróleo / Coca‑Cola) para cada día:", ratio_list)



Precios del barril de petróleo (últimos 7 días): [76.24, 77.88, 78.31, 77.22, 76.71, 76.45, 77.17]
Precio de un 'barril' de Coca‑Cola (159 litros): 238.5
Relación (Petróleo / Coca‑Cola) para cada día: [0.32, 0.327, 0.328, 0.324, 0.322, 0.321, 0.324]


Al analizar la relación entre el precio del barril de petróleo y el precio equivalente de un "barril" de Coca-Cola, se observa que la Coca-Cola es consistentemente más cara, con un precio promedio de $238.50 frente al rango de $76 a $78 del petróleo. Esto refleja cómo productos de consumo con altos costos de producción, comercialización y marca, como la Coca-Cola, pueden superar significativamente en valor a una materia prima esencial como el petróleo, que a pesar de su importancia global, tiene un precio determinado principalmente por factores de oferta y demanda en mercados internacionales.